core/services/ocr2/plugins/ocr2vrf/coordinator/coordinator.go

```diff
@@ -0,0 +1,426 @@
+			c.newTransmissionTopic,
+		},
+		c.coordinatorAddress,
+		pg.WithParentCtx(ctx))
+	if err != nil {
+		err = errors.Wrap(err, "logs with topics")
+		return
+	}
+
+	var (
+		randomnessRequestedLogs            []vrf_wrapper.VRFBeaconCoordinatorRandomnessRequested
+		randomnessFulfillmentRequestedLogs []vrf_wrapper.VRFBeaconCoordinatorRandomnessFulfillmentRequested
+		randomWordsFulfilledLogs           []vrf_wrapper.VRFBeaconCoordinatorRandomWordsFulfilled
+		newTransmissionLogs                []vrf_wrapper.VRFBeaconCoordinatorNewTransmission
+	)
+	for _, lg := range logs {
+		switch {
+		case bytes.Equal(lg.EventSig, c.randomnessRequestedTopic[:]):
+			unpacked, err2 := unmarshalRandomnessRequested(lg) // FIXME
+			if err2 != nil {
+				// should never happen
+				err = errors.Wrap(err2, "unmarshal RandomnessRequested log")
+			}
+			randomnessRequestedLogs = append(randomnessRequestedLogs, unpacked)
+		case bytes.Equal(lg.EventSig, c.randomnessFulfillmentRequestedTopic[:]):
+			unpacked, err2 := unmarshalRandomnessFulfillmentRequested(lg) // FIXME
+			if err2 != nil {
+				// should never happen
+				err = errors.Wrap(err2, "unmarshal RandomnessFulfillmentRequested log")
+			}
+			randomnessFulfillmentRequestedLogs = append(randomnessFulfillmentRequestedLogs, unpacked)
+		case bytes.Equal(lg.EventSig, c.randomWordsFulfilledTopic[:]):
+			unpacked, err2 := unmarshalRandomWordsFulfilled(lg) // FIXME
+			if err2 != nil {
+				// should never happen
+				err = errors.Wrap(err2, "unmarshal RandomWordsFulfilled log")
+			}
+			randomWordsFulfilledLogs = append(randomWordsFulfilledLogs, unpacked)
+		case bytes.Equal(lg.EventSig, c.newTransmissionTopic[:]):
+			unpacked, err2 := unmarshalNewTransmission(lg) // FIXME
+			if err2 != nil {
+				// should never happen
+				err = errors.Wrap(err2, "unmarshal NewTransmission log")
+			}
+			newTransmissionLogs = append(newTransmissionLogs, unpacked)
+		}
+	}
+
+	// Scan for blocks where an output is required
+	// blocksRequested maps block number to the block object.
+	type key struct {
+		blockNumber uint64
+		confDelay   uint32
+	}
+	blocksRequested := make(map[key]struct{})
+	for _, r := range randomnessRequestedLogs {
+		if _, ok := confirmationDelays[uint32(r.ConfDelay.Uint64())]; !ok {
+			// if we can't find the conf delay in the map then just ignore this request
+			c.lggr.Infow("ignoring bad request, unsupported conf delay",
+				"confDelay", r.ConfDelay.String(),
+				"supportedConfDelays", confirmationDelays)
+			continue
+		}
+		// If the next beacon output height is less than currentHeight - conf delay
+		// AND the log has enough confirmations, then we can schedule it to be fulfilled.
+		cond := r.ConfDelay.Int64() < currentHeight // TODO: is this necessary? Won't this always be true?
+		cond = cond && r.NextBeaconOutputHeight < uint64(currentHeight-r.ConfDelay.Int64())
+		cond = cond && currentHeight >= int64(r.Raw.BlockNumber+r.ConfDelay.Uint64()) // TODO: is this redundant?
```

Comment:
it does seem redundant since `r.NextBeaconOutputHeight` > `r.Raw.BlockNumber` ([code](https://github.com/smartcontractkit/threshold-vrf/blob/master/contracts/VRFBeaconExternalAPI.sol#L294))